snuba/optimize.py

```diff
@@ -72,12 +91,30 @@ def get_partitions_to_optimize(
     return parts
 
 
-def optimize_partitions(clickhouse, database, table, parts):
-    query_template = """\
+def optimize_partitions(
+    clickhouse: ClickhousePool,
+    storage_key: StorageKey,
+    database: str,
+    table: str,
+    parts: Sequence[util.Part],
+) -> None:
+
+    events_query_template = """\
         OPTIMIZE TABLE %(database)s.%(table)s
         PARTITION ('%(date_str)s', %(retention_days)s) FINAL
     """
 
+    errors_transactions_query_template = """\
+        OPTIMIZE TABLE %(database)s.%(table)s
+        PARTITION (%(retention_days)s, '%(date_str)s') FINAL
+    """
```

Comment:
The fact you need two separate templates when the only difference is the order of the fields seems hard to maintain when adding more storages or rebuilding tables.
It seems the need for two separate templates comes from the same issue I mentioned above about the lack of a structure for the partition key (that we could associate with the schema, or, better, take from the DB).

Reply:
This was actually unnecessary, removed this and passed the part names in